### PR TITLE
Add -orchardactionlimit parameter to guard against memory exhaustion.

### LINKED ISSUE #5888: Zcashd experiences an OOM-kill when sending a very large number of Orchard notes
In this case, my spend input was approximately 850 notes.

### LINKED ISSUE #5889: z_sendmany: include some kind of limit parameter with defaults based on memory and blocksize
Related to [#5888](https://github.com/zcash/zcash/issues/5888).

### PR DESCRIPTION
**Pull request by nuttycom**
Orchard proving can require large amounts of memory, so by default
`z_sendmany` will not attempt to create transactions containing more
than 50 Orchard inputs or outputs to reduce the risk of memory
exhaustion.  The `-orchardactionlimit` parameter allows users with
larger amounts of memory at their disposal to override this limit.

Fixes #5889
Fixes #5888

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ Option handling
 - The `-reindex` and `-reindex-chainstate` options now imply `-rescan`
   (provided that the wallet is enabled and pruning is disabled, and unless
   `-rescan=0` is specified explicitly).
+- A new `-orchardactionlimit` option has been added to allow the user to
+  override the default maximum of 50 Orchard actions per transaction. 
+  Transactions that contain large numbers of Orchard actions can use 
+  large amounts of memory for proving, so the 50-action default limit is
+  imposed to guard against memory exhaustion. Systems with more than 16G
+  of memory can safely set this parameter to allow 200 actions or more.
```
Comment (ebfull) [inline doc/release-notes.md:18]: How many actions in a tx can fit in a block? I had thought it was less than 200.

**Review by ebfull — APPROVED** [review]

**Review by str4d — DISMISSED** [review]
utACK, but this will conflict with #5917, and this PR is much smaller and thus easier to review post-rebase, so let's merge #5917 first.

File: doc/release-notes.md (hunk 1 above)
Comment (nuttycom) [reply doc/release-notes.md:18]: I was able to create a tx with 800 actions.

**Review by ebfull — APPROVED** [review]

File: src/wallet/wallet.h [hunk 2]
```diff
@@ -54,6 +54,9 @@ extern bool bSpendZeroConfChange;
 extern bool fSendFreeTransactions;
 extern bool fPayAtLeastCustomFee;
 extern unsigned int nAnchorConfirmations;
+// The maximum number of Orchard actions permitted within a single transaction.
+// This can be overridden with the -orchardactionlimit CLI flag.
```
Comment (LarryRuane) [inline src/wallet/wallet.h:58]: I don't think it's a CLI flag; it's a configuration option (command line or `zcash.conf`).

**Review by LarryRuane — CHANGES_REQUESTED** [review]
Should add an entry to `CWallet::GetWalletHelpString` (`orchardanchorconfirmations` is also missing an entry here, I didn't check if there are others).

Also, probably the case but just to be sure: verifying a transaction with many actions doesn't require a large amount of memory, does it? (That would be very bad if so!)

Otherwise looks good (I assume this PR consists of only the last commit, that you're layering this PR on some other unmerged PRs).

**Review by LarryRuane — APPROVED** [review]
utACK 20614fdaf8b3981f77ab57db699d4da7a97cfde6

**Review by str4d — APPROVED** [review]
utACK 20614fdaf8b3981f77ab57db699d4da7a97cfde6

**Review by daira — COMMENTED** [review]
Post-hoc utACK a71db41d38d9dcedc16eec2818c03e9234a9ed9a and 20614fdaf8b3981f77ab57db699d4da7a97cfde6

**Pull request merged**